### PR TITLE
Make content tracking work, add IP forwarding and fix travis CI runs

### LINKED ISSUE #51: Curl request should forward a detected IP and their headers
See https://github.com/matomo-org/matomo/issues/14691#issuecomment-514950013 and https://github.com/matomo-org/matomo/issues/14691#issuecomment-515336708

Eg it should forward `X-Forwarded-For` header.

### LINKED ISSUE #52: content tracking does not work when using the tracker-proxy
Tracking page views and links works as expected, but content tracking does not work.
I've verified that content tracking does work if I'm using the non proxy javascript tracker.

This is for matomo 3.11.0, php 7.1.31 and current matomo proxy tracker as a fresh clone via git clone.

The matomo instance is running on its own subdomain matomoinstance.mydomain.tld.
piwik.php is accessible through the path /stat/ on each website to be tracked.

Tracker code is:
```javascript
<script type="text/javascript">
  var _paq = window._paq || [];
  _paq.push(['disableCookies']);
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  _paq.push(['trackAllContentImpressions']);
  (function() {
    var u="//mysite.mydomain.tld/stat/";
    _paq.push(['setTrackerUrl', u+'piwik.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type="text/javascript"; g.async=true; g.defer=true; g.src=u+"piwik.php"; s.parentNode.insertBefore(g,s);
  })();
</script>
```

config.php for the proxy:
```php
<?php

$PIWIK_URL = 'https://mymatomoinstance.mydomain.tld/';
$PROXY_URL = '/stat/';
$TOKEN_AUTH = 'someworkingauthtoken';
$timeout = 5;
$user_agent = '';

```

### LINKED ISSUE #54: Update tests and readme to current php version, travis CI fails
The tests assume php 5.5 which is eol since more than 3 years and php 5.4 which is event not mentioned anymore at https://www.php.net/supported-versions.php

I think minimum php 7.2 should be used for testing

### PR DESCRIPTION
**Pull request by level420**
Forward raw POST body data instead of destructed data from ```$_POST``` array.
Add configurable IP forwarding header for forwarding the original client IP.
Fixes #51
Fixes #52
Fixes #54

### CODE REVIEW

**Comment by level420** [comment]
Tested with both fopen and curl.

**Comment by level420** [comment]
Look mom! even made travis happy!

File: proxy.php [hunk 1]
```diff
@@ -272,16 +273,20 @@ function getHttpContentAndStatus($url, $timeout, $user_agent)
...
         $stream_options['http']['content'] = $postBody;
+        
+        if(!empty($http_ip_forward_header)) {
+            $visitIp = getVisitIp();
+            $stream_options['http']['header'][] = "$http_ip_forward_header: $visitIp";
+        }
     }
 
     if($useFopen) {
-        $stream_options['http']['header'] = implode("\r\n", $header);
```
Comment (tsteur) [inline proxy.php]: just checking is this here done on purpose?

File: proxy.php [hunk 2]
```diff
@@ -272,16 +273,20 @@ function getHttpContentAndStatus($url, $timeout, $user_agent)
 
     // if there's POST data, send our proxy request as a POST
     if (!empty($_POST)) {
-        $postBody = http_build_query($_POST);
+        $postBody = file_get_contents("php://input");
```
Comment (tsteur) [inline proxy.php]: Looks good to me. @mattab and @diosmosis any thoughts maybe too?

Comment (diosmosis) [reply proxy.php]: Looks good to me too, except possibly for the other comment you made.

File: proxy.php (hunk 1 above)
Comment (level420) [reply proxy.php]: Yes. When using the implode method by concatenating all headers in one string it didn't work.
Starting from ca. php 5.2 ```stream_context_create``` accepts an array for multiple headers to be set.
See https://www.php.net/manual/en/context.http.php 
Therefore the implode line could be omitted.

It seems that the implode method is not enough as the header string needs an additional ```"\r\n"```.

https://stackoverflow.com/questions/17467601/php-http-request-stream-context-create
but why care if ```stream_context_create``` does it the right way if an array of headers is given.

Comment (level420) [reply proxy.php]: I've re-visited the php error log backwards before I made this change and found the following entry:
```
PHP Notice:  file_get_contents(): Content-type not specified assuming application/x-www-form-urlencoded in /matomo/proxy.php on line 288
```
which also indicates that the headers were not sent correctly before.

Comment (level420) [reply proxy.php]: Mmpf: hold it. It is OK as it is now, as the header is set here: https://github.com/matomo-org/tracker-proxy/blob/master/proxy.php#L257
So no need to change something.

**Comment by tsteur** [comment]
Very much appreciated @level420 👍

**Pull request merged**